network/p2p/middleware.go

```diff
@@ -135,6 +158,37 @@ func DefaultValidators(log zerolog.Logger, flowID flow.Identifier) []network.Mes
 	}
 }
 
+func (m *Middleware) topologyPeers() (peer.IDSlice, error) {
+	identities, err := m.ov.Topology()
+	if err != nil {
+		return nil, err
+	}
+
+	return m.peerIDs(identities.NodeIDs()), nil
+}
+
+func (m *Middleware) allPeers() peer.IDSlice {
+	return m.peerIDs(m.idProvider.Identifiers())
+}
+
+func (m *Middleware) peerIDs(flowIDs flow.IdentifierList) peer.IDSlice {
+	result := make([]peer.ID, len(flowIDs))
+
+	for _, fid := range flowIDs {
+		pid, err := m.idTranslator.GetPeerID(fid)
+		if err != nil {
+			// We probably don't need to fail the entire function here, since the other
+			// translations may still succeed
+			m.log.Err(err).Str("flowID", fid.String()).Msg("failed to translate to peer ID")
+			continue
```

Comment:
Just wanted to raise awareness here, I'm open to the argument that we actually should fail here.

Reply:
In particular, because the new behavior will likely differ a bit from the old behavior: https://github.com/onflow/flow-go/pull/1133/files#r691579930

I feel that the new behavior is actually better, but I'm not super confident on this.

Reply:
In general, my thinking was that returning a partial list of peer IDs is at least better than completely failing because a single flow ID couldn't be translated.

Reply:
yeah, dont need to fail - I had to change the return earlier on v0.20 and [master](https://github.com/onflow/flow-go/pull/1138)